src/test/common/test_type_check_decorator.py

```diff
@@ -153,3 +274,40 @@ def mix_args(a: DummyCustom1, *i: DummyCustom2):
             mix_args(dummy_1, [dummy_2, 1])
         assert "'i'[1] must be 'DummyCustom2'. The given argument type is 'int'"\
                in str(e.value)
+
+    def test_method_case(self):
+        class ValidateTestClass:
+
+            @type_check_decorator
+            def bool_arg(self, a: bool, b: bool):
+                pass
+
+        v = ValidateTestClass()
+
+        # Ensure that it can be executed at the appropriate input
+        v.bool_arg(a=True, b=False)
+
+        with pytest.raises(UnsupportedTypeError) as e:
+            v.bool_arg(True, 'test')
+        assert "'b' must be 'bool'. The given argument type is 'str'"\
+               in str(e.value)
+
+    def test_method_case_variable_length_arg(self):
+        class ValidateTestClass:
+
+            @type_check_decorator
+            def bool_arg(self, *i: DummyCustom1):
```

Comment:
The function name is not suitable and should be changed.
```suggestion
            def var_len_args(self, *i: DummyCustom1):
```

Reply:
fixed